tests/python_package_test/test_engine.py

```diff
@@ -163,6 +163,9 @@ def test_cv(self):
         tss = TimeSeriesSplit(3)
         lgb.cv({'verbose': -1}, lgb_train, num_boost_round=10, data_splitter=tss, nfold=5,  # test if wrong nfold is ignored
                metrics='l2', verbose_eval=False)
+        # params has a string type metric and metrics is not None
+        lgb.cv({'verbose': -1, 'metric': 'binary_logloss'}, lgb_train, num_boost_round=10, nfold=3,
+               shuffle=False, metrics='auc', verbose_eval=False)
```

Comment:
the default task is regression, so metrics better be l1 or l2. I think you can re-use L156-157